nucleus/common/jul-extensions/src/main/java/fish/payara/logging/jul/PayaraLogManager.java

```diff
@@ -0,0 +1,730 @@
+ * <li>Or you can use {@link #reconfigure(PayaraLogManagerConfiguration, Action, Action)} method,
+ * which provides a way to add programatical extension points.
+ * </ul>
+ * <p>
+ * Note: Some methods have complicated implementation, the reason is that JDK {@link LogManager} is
+ * not an example of well usable and well extensible class at all.
+ *
+ * @author David Matejcek
+ */
+public class PayaraLogManager extends LogManager {
+
+    /** Property key for a list of root handler implementations */
+    public static final String KEY_ROOT_HANDLERS = "handlers";
+    /** Property key for a level of system root logger. System root loggers children are not configurable. */
+    public static final String KEY_SYS_ROOT_LOGGER_LEVEL = "systemRootLoggerLevel";
+    /** Property key for a level of user root logger. User root loggers children can have own level.  */
+    public static final String KEY_USR_ROOT_LOGGER_LEVEL = ".level";
+    /** Empty string - standard root logger name */
+    public static final String ROOT_LOGGER_NAME = "";
+
+    private static volatile PayaraLoggingStatus status = PayaraLoggingStatus.UNINITIALIZED;
+    private static PayaraLogManager payaraLogManager;
+    private static final AtomicBoolean protectBeforeReset = new AtomicBoolean(true);
+
+    private volatile PayaraLogger systemRootLogger;
+    private volatile PayaraLogger userRootLogger;
+    private volatile PayaraLogger globalLogger;
+
+    private PayaraLogManagerConfiguration configuration;
+
+
+    static synchronized boolean initialize(final Properties configuration) {
+        PayaraLoggingTracer.trace(PayaraLogManager.class, "initialize(configuration)");
+        if (status.ordinal() > PayaraLoggingStatus.UNINITIALIZED.ordinal()) {
+            PayaraLoggingTracer.error(PayaraLogManager.class,
+                "Initialization of the logging system failed - it was already executed");
+            return false;
+        }
+        // We must respect that LogManager.getLogManager()
+        // - creates final root and global loggers,
+        // - calls also addLogger.
+        // - calls setLevel if the level was not set in addLogger.
+        // OR something already configured another log manager implementation
+        // Therefore status is now moved directly to UNCONFIGURED
+        status = PayaraLoggingStatus.UNCONFIGURED;
+        final PayaraLogManager logManager = getLogManager();
+        if (logManager == null) {
+            // oh no, another LogManager implementation is already used.
+            return false;
+        }
+        logManager.doFirstInitialization(ensureSortedProperties(configuration));
+        return true;
+    }
+
+
+    /**
+     * @return true if {@link PayaraLogManager} is configured as the JVM log manager.
+     */
+    public static boolean isPayaraLogManager() {
+        if (payaraLogManager == null) {
+            return getLogManager() != null;
+        }
+        return payaraLogManager != null;
```

Comment:
Test not needed, always true.

Reply:
Aha, that last line, you are right, (I hate cyclic dependencies, that is how it came to this, I did not notice it is meaningless now)
Thanks! :+1: